src/Encamina.Enmarcha.Agents/Telemetry/TelemetryAgentIdInitializer.cs

```diff
@@ -14,20 +12,22 @@ namespace Encamina.Enmarcha.Agents.Telemetry;
 /// </summary>
 public class TelemetryAgentIdInitializer : ITelemetryInitializer
 {
+    private static readonly AsyncLocal<AgentTelemetryContext> CurrentContext = new();
```

Comment:
We’re setting this to `null` in `ClearCurrent`, so maybe `CurrentContext` should be nullable `AsyncLocal<AgentTelemetryContext?>` instead?